engine/apps/schedules/models/custom_on_call_shift.py

```diff
@@ -433,6 +433,34 @@ def get_rotation_date(self, event_ical, get_next_date=False):
             return
         return next_event_dt
 
+    def get_last_event_date(self, date):
+        """Get start date of the last event before the chosen date"""
+        assert date >= self.start, "Chosen date should be later or equal to initial event start date"
+
+        event_ical = self.generate_ical(self.start)
+        initial_event = Event.from_ical(event_ical)
+        # take shift interval, not event interval. For rolling_users shift it is not the same.
+        interval = self.interval or 1
+        if "rrule" in initial_event:
+            # means that shift has frequency
+            initial_event["rrule"]["INTERVAL"] = interval
+        initial_event_start = initial_event["DTSTART"].dt
+
+        last_event = None
+        # repetitions generate the next event shift according with the recurrence rules
+        repetitions = UnfoldableCalendar(initial_event).RepeatedEvent(
+            initial_event, initial_event_start.replace(microsecond=0)
+        )
+        ical_iter = repetitions.__iter__()
+        for event in ical_iter:
```

Comment:
can probably remove `ical_iter` variable 
```suggestion
        for event in repetitions.__iter__():
```